Purchases/Identity/CustomerInfoManager.swift

```diff
@@ -174,9 +166,35 @@ class CustomerInfoManager {
         }
     }
 
+    @available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.0, *)
+    var customerInfoStream: AsyncStream<CustomerInfo> {
+        return AsyncStream(bufferingPolicy: .bufferingNewest(1)) { continuation in
+            let disposable = self.monitorChanges { continuation.yield($0) }
+
+            continuation.onTermination = { @Sendable _ in disposable() }
+        }
+    }
+
+    private var customerInfoObserversByIdentifier: [Int: (CustomerInfo) -> Void] = [:]
+
+    /// Allows monitoring changes to the active `CustomerInfo`.
+    /// - Returns: closure that removes the created observation.
+    /// - Note: this method is not thread-safe.
+    func monitorChanges(_ changes: @escaping (CustomerInfo) -> Void) -> () -> Void {
+        let identifier = self.customerInfoObserversByIdentifier.keys
+            .sorted().last.map { $0 + 1 } // Next index
+            ?? 0 // Or default to 0
+
```

Comment:
This may just be me, but it took me a bit to understand what was going on here. Perhaps we can make it a little clearer by doing less at once? Something like
```suggestion
        let currentIdentifier = self.customerInfoObserversByIdentifier.keys
            .sorted()
            .last
        let nextIdentifier = currentIdentifier ? currentIdentifier + 1 : 0
```

Reply:
I'll split it 👍 
I'll still use `.map` + `??` to avoid the force unwrap though.